BotTest/NameMatchTest.cs

```diff
@@ -0,0 +1,148 @@
+using Discord;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using TU20Bot;
+using TU20Bot.Commands;
+using TU20Bot.Configuration;
+
+namespace BotTest
+{
+    [TestClass]
+    public class NameMatchTest
+    {
+
+        private static NameMatch _nameMatch;
+        private static Config _config;
+        private static Client _client;
+        private static Handler _handler;
+
+        private const bool OUTPUT_TO_DISCORD = false;
+
+        [ClassInitialize]
+        public static async Task ClassInitialize(TestContext testContext)
+        {
+
+            string token = "";
+            testContext.WriteLine("Iteration 1");
+
+            // Start bot with token from "token.txt" in working folder.
+            try
+            {
+                var _token = File.ReadAllText("../../../../token.txt").Trim();
```

Comment:
No, this relative reference is so that we don't need to add a duplicate or link of `token.txt` into the `BotTest` project. This line will need to be tested in a Windows environment as they use a backslash separator for paths.

Reply:
Windows can use either `/` or `\`, no need to worry.

I don't really see how this is effective. If you want to specify a location to store both token.txts you change the current directory in the project settings. This is workaroundy and takes away the ability to do the above.

Reply:
> Windows can use either `/` or `\`

That's great! I have also confirmed this on my test system.

In terms of efficiency, this solution isn't great as it adds an extra maintenance step if the test output directory ever changes. Thanks for bringing notice to this line, I found a simple solution where we can automatically link `token.txt` into the project directory by specifying the relationship within the `csproj` file.